packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing-tile.tsx

```diff
@@ -0,0 +1,603 @@
+      this.nextBillingDate = formatDate(
+        serviceData.billing?.nextBillingDate,
+        this.language,
+      );
+      if (
+        serviceData.billing.engagement === null &&
+        !serviceData.billing.engagementRequest
+      ) {
+        // should be null if no commitment
+        // Red chip 'None', no link or link 'Commit'
+        this.commitmentStatus = 'none';
+      } else if (serviceData.billing.engagement?.endDate < new Date()) {
+        // Red chip 'Ended DATE', no link
+        this.commitmentStatus = 'ended';
+      } else if (serviceData.billing.engagement?.endDate > new Date()) {
+        // No chip 'Renews on DATE', link 'Re-commit and get a discount'
+        this.commitmentStatus = 'renews';
+      } else if (serviceData.billing.engagementRequest) {
+        // No chip 'Your service commitment will begin from DATE'
+        this.commitmentStatus = 'requested';
+        this.requestDate = serviceData.billing.engagementRequest.requestDate;
+      } else if (serviceData.billing.expirationDate) {
+        // No chip 'Ends DATE'
+        this.commitmentStatus = 'ends';
+      }
+    });
+    if (this.getServiceType() === 'DOMAIN')
+      await fetchDomainOwner(this.getServiceName()).then((serviceData) => {
+        this.whoisOwnerDomain = serviceData.whoisOwner;
+      });
+  }
+
+  getServiceName() {
+    const parts = this.servicePath.split('/');
+    return parts[parts.length - 1]; // get the last part
+  }
+
+  getServiceType() {
+    const parts = this.servicePath.split('/');
+    parts.pop(); // Remove the service name
+    return parts.join('_').toUpperCase(); // Join remaining parts with '_', and convert to upper case
+  }
+
+  render() {
+    const RenewalContent = () => {
+      const status = this.renewStatus;
+      const getColor = () => {
+        switch (status) {
+          case 'deleteAtExpiration':
+            return OdsThemeColorIntent.error;
+          case 'automatic':
+            return OdsThemeColorIntent.success;
+          case 'manualPayment':
+            return OdsThemeColorIntent.warning;
+          case 'expired':
+            return OdsThemeColorIntent.error;
+          default:
+            return OdsThemeColorIntent.default;
+        }
+      };
+
+      const getText = () => {
+        switch (status) {
+          case 'deleteAtExpiration':
+            return this.localStrings
+              .manager_billing_service_status_delete_at_expiration;
+          case 'automatic':
+            return this.localStrings.manager_billing_service_status_automatic;
+          case 'manualPayment':
+            return this.localStrings
+              .manager_billing_service_status_manualPayment;
+          case 'expired':
+            return this.localStrings.manager_billing_service_status_expired;
+          default:
+            return this.localStrings[
+              `manager_billing_service_status_${status}`
+            ];
+        }
+      };
+
+      const getMenuLink = () => {
```

Comment:
can you split it in other file menu please ?

Reply:
Done